notify/impl.go

```diff
@@ -613,15 +617,21 @@ func (n *PagerDuty) notifyV2(
 
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return false, err
+		return false, fmt.Errorf("failed to encode PagerDuty v2 message: %v", err)
 	}
 
 	resp, err := post(ctx, c, n.conf.URL.String(), contentTypeJSON, &buf)
 	if err != nil {
-		return true, err
+		return true, fmt.Errorf("failed to post message to PagerDuty: %v", err)
 	}
 	defer resp.Body.Close()
 
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return true, fmt.Errorf("failed to parse response from PagerDuty: %v", err)
+	}
+	level.Debug(n.logger).Log("msg", fmt.Sprintf("Received response from PagerDuty(status: %d): %s", resp.StatusCode, string(body)), "incident", key)
```

Comment:
Maybe log only when PagerDuty replies with 400? Looking at https://v2.developer.pagerduty.com/docs/events-api-v2#api-response-codes--retry-logic the retry logic implemented by the AlertManager isn't correct as we retry on all responses other than `2XX`.

Reply:
Fixed.
Are you sure the retry logic is wrong? https://github.com/prometheus/alertmanager/blob/ec7b4686fdfab2d530407ce28250e29a1e7e8178/notify/impl.go#L712-L714
Looks like we're not retrying on 400.

Reply:
Sorry I somehow misread the code...